pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+const (
+	// containerNamePrefix is used to identify the containers/sandboxes on the node managed by kubelet
+	containerNamePrefix = "k8s"
+
+	// Taken from lmctfy https://github.com/google/lmctfy/blob/master/lmctfy/controllers/cpu_controller.cc
+	minShares     = 2
+	sharesPerCPU  = 1024
+	milliCPUToCPU = 1000
+
+	// 100000 is equivalent to 100ms
+	quotaPeriod    = 100000
+	minQuotaPeriod = 1000
+)
+
+// buildPodName creates a name which can be reversed to identify pod full name
+// This function returns stable name, unique name and an unique id.
+func buildPodName(podName, podNamespace, podUID string) (string, string, string) {
+	return buildContainerName(podName, podNamespace, podUID, nil)
+}
+
+// parsePodName unpacks a pod full name, returning the pod name, namespace and uid
+func parsePodName(name string) (string, string, string, error) {
+	podName, podNamespace, podUID, _, _, err := parseContainerName(name)
+	return podName, podNamespace, podUID, err
+}
+
+// buildContainerName creates a name which can be reversed to identify both full pod name and container name.
+// This function returns stable name, unique name and an unique id.
+func buildContainerName(podName, podNamespace, podUID string, container *api.Container) (string, string, string) {
+	// Build name for pod sandbox if container is nil
```

Comment:
I think we should make it explicitly what logic is used to build sandbox names and what for container names. I prefer create a different function for each of them.